src/components/CountrySelector.tsx

```diff
@@ -32,19 +33,41 @@ type CountrySelectorProps = {
 function CountrySelector({errorText = '', value: countryCode, onInputChange = () => {}, onBlur}: CountrySelectorProps, ref: ForwardedRef<View>) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const route = useRoute();
+    const countryFromUrlTemp = (route?.params as Record<string, string>)?.country;
```

Comment:
`route.params` are typed?

Reply:
How about: 
```ts
    import {SettingsNavigatorParamList} from '@libs/Navigation/types';
    import SCREENS from '@src/SCREENS';
    const route = useRoute<RouteProp<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.PROFILE.ADDRESS>>();
    const countryFromUrlTemp = route?.params?.country;
    const countryFromUrl = countryFromUrlTemp && CONST.ALL_COUNTRIES[countryFromUrlTemp] ? countryFromUrlTemp : '';
```

Reply:
Or better, how about moving this logic to a hook?

Reply:
Done, thanks!